alica_engine/include/engine/constraintmodul/Query.h

```diff
@@ -169,8 +169,8 @@ bool Query::getSolution(std::shared_ptr<const RunningPlan> rp, std::vector<Resul
         }
         // Again, the queried domain variables should be at the beginning of the domain variable segment
         // So a simple move and resize should do the trick
-        for (int i = 0; i < _queriedDomainVariables.size(); ++i) {
-            for (int j = domOffset; j < _relevantVariables.size(); ++j) {
+        for (int i = 0; i < static_cast<int>(_queriedDomainVariables.size()); ++i) {
```

Comment:
wouldn't it make more sense to declare `i` as whichever type .size() returns in the first place?

Reply:
well...there are different opinions about it.
Here's mine:
- size() returns size_t, i.e., uint64_t, and for performance reason i'd rather use a 32 bit counter.
- Imo, using an unsigned value for a size is a bad idea, and should be rectified whenever possible. Google's guideline gives a good explanation:
https://google.github.io/styleguide/cppguide.html#Integer_Types


tl;dr: I stubbornly fight size_t as a container size value.